src/components/Search/SearchDropdown.tsx

```diff
@@ -14,30 +14,41 @@ import {
   FontsRaw,
   Colors,
 } from '../../styles';
-import { InputType, ThemeMode, View } from '../../types/state';
+import { InputType, ThemeMode, View, ComparisonType } from '../../types/state';
 
 interface SearchDropdownProps {
   view: View;
   selectLabel: string;
   tooltipText: string;
   mode: ThemeMode;
   searchType: InputType;
+  comparisonType: ComparisonType;
 }
 
 function SearchDropdown({
   view,
   selectLabel,
   mode,
   searchType,
+  comparisonType,
 }: SearchDropdownProps) {
   const size = view == compareView ? 'small' : undefined;
   const { handleChangeDropdown } = useHandleChangeDropdown();
-  const searchState = useAppSelector((state) => state.search[searchType]);
+  let searchState = useAppSelector(
+    (state) => state.searchCompareWithBase[searchType],
+  );
+  if (comparisonType == 'searchCompareOverTime') {
+    searchState = useAppSelector((state) => state.searchCompareOverTime.new);
+  }
   const { repository } = searchState;
```

Comment:
The selector above should return `repository` directly instead of the full `searchState`.